taichi/backends/vulkan/spirv_ir_builder.cpp

```diff
@@ -34,6 +25,10 @@ void IRBuilder::init_header(bool support_int8,
 
   // capability
   ib_.begin(spv::OpCapability).add(spv::CapabilityShader).commit(&header_);
+  // FIXME: What about devices don't support this?
+  ib_.begin(spv::OpCapability)
```

Comment:
That's a good question. Can we guard this somehow (at least by SPIR-V version?) We do need to put mobile platforms in mind here.